test/scheduleRecurringMetricsGeneration.test.ts

```diff
@@ -0,0 +1,78 @@
+import 'mocha'
+import { assert } from 'chai'
+
+import { AddressCount, Metrics } from '../src'
+
+import config from './config'
+import structuredClone from './helpers'
+
+let metrics: Metrics
+
+describe('Push Metrics - scheduleRecurringMetricsGeneration()', function (): void {
+  afterEach('', function () {
+    metrics.stopMetrics()
+  })
+
+  it('publishes all gauges', async function () {
+    const expectedPayIdCount = 5
+    const expectedBtcAddressCount = 3
+    const expectedXrpAddressCount = 4
+    const metricsConfig = structuredClone(config)
+    const addressCountFn = async (): Promise<AddressCount[]> => {
+      return [
+        {
+          paymentNetwork: 'BTC',
+          environment: 'MAINNET',
+          count: expectedBtcAddressCount,
+        },
+        {
+          paymentNetwork: 'XRPL',
+          environment: 'MAINNET',
+          count: expectedXrpAddressCount,
+        },
+      ]
+    }
+
+    const payIdCountFn = async (): Promise<number> => expectedPayIdCount
+
+    metrics = new Metrics(metricsConfig, addressCountFn, payIdCountFn)
+    await metrics.generateAddressCountMetrics()
+    await metrics.generatePayIdCountMetrics()
+    metrics.scheduleRecurringMetricsGeneration()
+
+    assertMetrics(
+      /payid_server_info\{org="example.com",serverAgent="unittest:1.2.3",protocolVersion="1.0.0"\} 1/u,
+    )
+
+    assertMetrics(
+      new RegExp(
```

Comment:
nit: Why do you use regex literals sometimes and the `RegExp()` constructor in other places?

Reply:
Because the linter. I have to use `new RegExp()` when the regex uses variables, but not when the regex is static text. The linter complains if I used `new RegExp()` with static text.